components/eamxx/docs/developer/code_structure.md

```diff
@@ -0,0 +1,154 @@
+    │   ├── physics/
+    │   │   ├── cld_fraction/
+    │   │   ├── cosp/
+    │   │   ├── iop_forcing/
+    │   │   ├── mam/
+    │   │   ├── ml_correction/
+    │   │   ├── nudging/
+    │   │   ├── p3/
+    │   │   ├── register_physics.hpp
+    │   │   ├── rrtmgp/
+    │   │   ├── share/
+    │   │   │   ├── physics_constants.hpp
+    │   │   │   └── physics_functions.hpp
+    │   │   ├── shoc/
+    │   │   ├── spa/
+    │   │   └── tms/
+    │   ├── python/
+    │   ├── eamxx_config.h.in
+    │   └── share/
+    ├── tests/
+    │   ├── generic/
+    │   ├── meta-tests/
+    │   ├── multi-process/
+    │   │   ├── dynamics_physics/
+    │   │   └── physics_only/
+    │   ├── python/
+    │   └── single-process/
+    └── tpls/
+    ```
+
+<!-- <div class="grid" markdown> -->
+=== "EAMxx Root Directory"
+    All EAMxx-specific code can be found in `components/eamxx` within the
+    [E3SM repo](https://github.com/E3SM-Project/E3SM).
+    Here's how things are organized:
+
+    - `cime_config`: Tools and XML files for integrating EAMxx with E3SM via the
+      CIME framework.
+    - `cmake`: CMake functions and macros used by the configuration/build system.
+    - `docs`: Documentation for the EAMxx project, including design documents,
+      instructions for building and testing EAMxx, and this document.
+    - `scripts`: Miscellaneous scripts that implement workflows for running tests
+      and analyzing performance.
+    - `src`: All C++ source code (and any bridges to Fortran) for EAMxx are stored
+      here. We describe the contents of this directory in greater detail below.
+    - `tests`: Implements standalone, end-to-end tests for various EAMxx
+      components (RRTMG, HOMME, P3, SHOC, etc).
+    - `tpls`: Utilities for building EAMxx-specific third-party libraries.[^arcane]
+    [^arcane]:
+        *Here there be dragons...* These are some of the more arcane EAMxx
+        features that even the bravest of developers may never lay eyes on.
+    
+    In addition, you'll notice the following files in `components/eamxx`:
+    
+    - `CMakeLists.txt`: The CMake file that defines EAMxx's configuration/build
+      system.
+    - `CTestConfig.cmake`: This CTest file contains parameters that determine how
+      our test results are reported to the [E3SM CDash Site](http://my.cdash.org/submit.php?project=E3SM).
+    - `README.md`: EAMxx's top-level README file, which describes the project and
+      its purpose.
+    - `mkdocs.yml`: The configuration file for [mkdocs](https://www.mkdocs.org/),
+      the tool we currently use to build and publish our documentation.
+
+=== "`src/` Directory"
+
+    Herein lies the source code for EAMxx.
+    Broadly, here's where things are located:
+    
+    - `control`: Contains the atmosphere driver and basic tests for it.
+    - `diagnostics`: A collection of simple classes used to compute diagnostic
+      quantities.
+    - `dynamics`: Here's where HOMME lives within EAMxx, along with code for
```

Comment:
More precisely, the files implementing the necessary interfaces between EAMxx and the HOMME non-hydrostatic dycore. HOMME itself is in components/homme